contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -1234,6 +1235,154 @@ uint64_t Filter::getMergedConfigId(ProcessorState& state) {
   return id;
 }
 
+void Filter::defineMetricCommon(uint32_t metric_type, absl::string_view name, uint32_t* metric_id) {
+  auto type = static_cast<MetricType>(metric_type);
+
+  Stats::StatNameManagedStorage storage(name, config_->metricStore().scope_->symbolTable());
+  Stats::StatName stat_name = storage.statName();
+  if (type == MetricType::Counter) {
+    auto id = config_->metricStore().nextCounterMetricId();
+    auto c = &config_->metricStore().scope_->counterFromStatName(stat_name);
+    config_->metricStore().counters_.emplace(id, c);
+    *metric_id = id;
+  } else if (type == MetricType::Gauge) {
+    auto id = config_->metricStore().nextGaugeMetricId();
+    auto g = &config_->metricStore().scope_->gaugeFromStatName(
+        stat_name, Stats::Gauge::ImportMode::Accumulate);
+    config_->metricStore().gauges_.emplace(id, g);
+    *metric_id = id;
+  } else { // (type == MetricType::Histogram) {
+    auto id = config_->metricStore().nextHistogramMetricId();
+    auto h = &config_->metricStore().scope_->histogramFromStatName(
+        stat_name, Stats::Histogram::Unit::Unspecified);
+    config_->metricStore().histograms_.emplace(id, h);
+    *metric_id = id;
+  }
+}
+
+void Filter::incrementMetricCommon(uint32_t metric_id, int64_t offset) {
+  auto type = static_cast<MetricType>(metric_id & MetricStore::kMetricTypeMask);
+  if (type == MetricType::Counter) {
+    auto it = config_->metricStore().counters_.find(metric_id);
+    if (it != config_->metricStore().counters_.end()) {
+      if (offset > 0) {
+        it->second->add(offset);
+      }
+    }
+  } else if (type == MetricType::Gauge) {
+    auto it = config_->metricStore().gauges_.find(metric_id);
+    if (it != config_->metricStore().gauges_.end()) {
+      if (offset > 0) {
+        it->second->add(offset);
+      } else {
+        it->second->sub(-offset);
+      }
+    }
+  }
+}
+
+void Filter::getMetricCommon(uint32_t metric_id, uint64_t* value) {
```

Comment:
nit: this method and `incrementMetricCommon` above seem to have a lot in common. Consider refactoring these

Reply:
these apis have been moved from filter to filter config.